packages/ra-ui-materialui/src/form/Toolbar.tsx

```diff
@@ -184,6 +184,7 @@ const Toolbar: FC<ToolbarProps> = props => {
                                       button.props.undoable,
                                       undoable
                                   ),
+                                  ...button.props,
```

Comment:
I don't understand this line. When we clone `button`, we clone it wit hits props. some are overridden starting at line 164, but when there is a possibility to set it directly on the button element, we use the `valueOrDefault` util. I think this is what you should do for the `resource` prop.